src/core/deltafilewrapper.cpp

```diff
@@ -892,17 +778,187 @@ void DeltaFileWrapper::addCreate( const QString &localLayerId, const QString &so
 
   delta.insert( QStringLiteral( "new" ), newData );
 
-  QString localPk = delta.value( QStringLiteral( "localPk" ) ).toString();
-  mLocalPkDeltaIdx[localLayerId][localPk] = static_cast<int>( mDeltas.count() );
+  appendDelta( delta );
+}
 
-  mDeltas.append( delta );
-  mIsDirty = true;
+void DeltaFileWrapper::appendDelta( const QJsonObject &delta )
+{
+  mergeDelta( delta );
+}
```

Comment:
Not clear what is the purpose of this method, if it is just a renaming for `mergeDelta`? 

Isn't the goal "always merge deltas unless deltafile wrapper is busy". If so, then `appendDelta` is enough and whether we merge or append to the deltafile is decided within `appendDelta`, not by the caller.

Reply:
@suricactus , doh, big part missing here.

```
if ( mIsPushing )
{
  mPendingDeltas << delta;
}
else
{
  mergeDelta( delta );
}
```

I thought it'd be better to regroup that logic into one function instead of having it three times in add{Create,Delete,Patch}().